src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -149,23 +162,30 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
         // check if already checked and passed the file
         if (!ordinaryChecks.isEmpty() || !commentChecks.isEmpty()) {
             final FileContents contents = getFileContents();
-            final DetailAST rootAST = JavaParser.parse(contents);
-            if (!ordinaryChecks.isEmpty()) {
-                walk(rootAST, contents, AstState.ORDINARY);
-            }
-            if (!commentChecks.isEmpty()) {
-                final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
-                walk(astWithComments, contents, AstState.WITH_COMMENTS);
-            }
-            if (filters.isEmpty()) {
-                addViolations(violations);
+            try {
+                final DetailAST rootAST = JavaParser.parse(contents);
+                if (!ordinaryChecks.isEmpty()) {
+                    walk(rootAST, contents, AstState.ORDINARY);
+                }
+                if (!commentChecks.isEmpty()) {
+                    final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
+                    walk(astWithComments, contents, AstState.WITH_COMMENTS);
+                }
+                if (filters.isEmpty()) {
+                    addViolations(violations);
+                }
+                else {
+                    final SortedSet<Violation> filteredViolations =
+                            getFilteredViolations(file.getAbsolutePath(), contents, rootAST);
+                    addViolations(filteredViolations);
+                }
+                violations.clear();
             }
-            else {
-                final SortedSet<Violation> filteredViolations =
-                    getFilteredViolations(file.getAbsolutePath(), contents, rootAST);
-                addViolations(filteredViolations);
+            catch (CheckstyleException ex) {
+                if (!skipFileOnJavaParseException) {
+                    throw ex;
+                }
```

Comment:
@romani We are not going to display anything when exception occurs? We are going to **completely** hide that this happened? Even with `haltOnException`, we display a message to the user to notify this happened.
https://github.com/checkstyle/checkstyle/blob/86195e06404df68905c90232f71a1c88491f9ad8/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L349C61-L365

This seems very risk to completely ignore it.

Reply:
>We are not going to display anything when exception occurs? We are going to completely hide that this happened?

If user explicitly configure to skip, we do skip swallow and no noise warning or any other messages. User should use it with caution.

Reply:
> I am for displaying a violation which can be suppressed, similar to halt on exception.

Yep, same here. Otherwise, how can we easily determine what is being skipped?

Reply:
If user use this option, he is completely ok to skip checkstyle on some classes.  Some Users value new syntax features more than checkstyle validation, it is normal, we should allow this.

It is same as all plugins have `skip` options, to allow keep plugin plugin still in build but let users decide when to activate it to fail a build or ignore it when they want this.

Reply:
> If user use this option, he is completely ok to skip checkstyle on some classes.

I’m not debating this; it is the essence of this feature. My question is how can a user *know* what is being skipped?

Reply:
No way, we do not provide any other communication message to users other than violations. Do we want to place violation with info level on whole file. But if we do this, we need to give user ability to adjust severity of violation as they want, even they might want to put "ignore". But it is extension of public config/design, I don't have good ideas for now, I would wait for feedback from users on how they use it in real life.

Reply:
We can just follow the existing model of haltOnException, we have no need to reinvent the wheel.

Reply:
https://github.com/checkstyle/checkstyle/blob/174bccc47ad62dae2b28c3fd10baf3a20809d425/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L351-L369

We do conversion of exception to violation by haltOnException.
Default severity is error 
https://github.com/checkstyle/checkstyle/blob/174bccc47ad62dae2b28c3fd10baf3a20809d425/src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java#L43

We need some property that do parse exception handling only and second property to define severity. @nrmancuso , can you suggest good names ?

Reply:
>second property to define severity

This is a really good idea. How about:
```xml
<module name="Checker">
  <property name="skipFileOnJavaParseException" value="true"/>
  <property name="parseExceptionSeverity" value="warning"/>
  <module name="TreeWalker">
    <module name="OverloadMethodsDeclarationOrder"/>
  </module>
</module>
```

Reply:
Sure, I have already come up with a functional workaround for the try-catch.

I would like to conclude everything here. I will print a long when there is a Java parsing error. AIso, a new property named `parseExceptionSeverity` will be added to control the log serverity.

For the new severity, do we prefer to open a new PR or just keep everything in here?

Reply:
@nrmancuso @romani @rnveach
According to https://github.com/checkstyle/checkstyle/pull/14779#discussion_r1573101500, do we move these two new properties to `Checker` instead of `TreeWalker`?

If we log the message about Java parsing exception in TreeWalker, there is still another layer of try-catch in `Checker`, which will print the exception information again. Therefore, just want to confirm that, we are discarding all the changes in `TreeWalker` and move what is here now into `Checker`?

Reply:
> we log the message 

It sounds like there may be some confusion, especially on log.

If `skipFileOnJavaParseException` is turned on, we are talking about writing the exception as a violation and never throwing the exception. The severity of the violation will be determined by `parseExceptionSeverity`.
If `skipFileOnJavaParseException` is turned off, we follow default behavior by throwing the exception. We let what other mechanisms we have take over as that is the default behavior.

Unless I am missing something, `skipFileOnJavaParseException` is really only about stopping the exception from skipping other `TreeWalker` processing.

Reply:
@Lmh-java we want to emulate the `haltOnException` property. It is not clear to me why checker should care. Parsing exception happens, we catch it and log the message. Is this behavior not possible?

Reply:
>According to https://github.com/checkstyle/checkstyle/pull/14779#discussion_r1573101500, do we move these two new properties to Checker instead of TreeWalker?

This is my mistake, these should live in TreeWalker for sure, thanks for calling this out.

Reply:
> Is this behavior not possible?

I think this is possible. let me try. :)

Reply:
That's a great idea. Thanks for your advice. I didn't think about `Optional` before. I will use that instead.